File: pset6/mario/mario.py
import cs50


def main():
    # Assign the range of input
    while True:
        n = cs50.get_int("Height: ")
        if n >= 1 and n <= 8:
            break

# print("#" if i + j >= n - 1 else " ", end="")
    # Create each row and hash
    for i in range(n):
        # First pyramid
        for j in range(n):
            if i + j >= n - 1:
                print("#", end="")
            else:
                print(" ", end="")
        # The gap between two pyramid
        for k in range(n - 2, n, 1):
            if k < n:
                print(" ", end="")
        # Second pyramid
        for l in range(n):
            if l <= i:
                print("#", end="")
        print()


main()

File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i ++)
    {
        for (int j = 0; j <= n - i; j++)
        {
            printf(" ");
        }
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        for (int l = 0; l < 2; l++)
        {
            printf(" ");
        }
        for (int m = 0; m <= i; m++)
        {
            printf("#");
        }
        printf("\n");
    }
}